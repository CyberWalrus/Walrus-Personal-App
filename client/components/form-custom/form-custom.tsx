import withFormState from "@client/hocs/with-authorization-state/with-authorization-state";
import { FormType, InputCustomOptions as options } from "@config/constants";
import * as React from "react";
import { ComponentClass, Fragment, FunctionComponent, ReactElement } from "react";
import { compose } from "redux";
import InputCustom from "../input-custom/input-custom";

interface PropsInsert {
  titel: string;
  formType: FormType;
}
interface PropsHoc {
  option: Option;
}
interface Option {
  email: string;
  password: string;
  formErrors: object;
  formValid: boolean;
  onChangeUserInput: () => void;
  onClickSubmit: () => void;
}
type Props = PropsHoc & PropsInsert;
const FormCustom: FunctionComponent<Props> = ({
  titel,
  option,
}: Props): ReactElement => {
  const {
    email,
    password,
    formErrors,
    formValid,
    onChangeUserInput,
    onClickSubmit,
  }: Option = option;
  const inputs: ReactElement[] = [];
  inputs.push(
    <InputCustom
      key={options.email.key}
      option={options.email}
      value={email}
      onChangeInput={onChangeUserInput}
    />,
  );
  inputs.push(
    <InputCustom
      key={options.password.key}
      option={options.password}
      value={password}
      onChangeInput={onChangeUserInput}
    />,
  );
  return (
    <form onSubmit={onClickSubmit} className={`form-custom`}>
      <div className={`form-custom__titel`}>
        <h2>{titel}</h2>
      </div>
      <div className={`form-custom__message`}>
        {formErrors &&
          Object.keys(formErrors).map(
            (fieldName: string, i: number): ReactElement => {
              if (formErrors[fieldName].length > 0) {
                return <p key={i}>{formErrors[fieldName]}</p>;
              } else {
                return <Fragment key={i} />;
              }
            },
          )}
      </div>
      {inputs.map((item: ReactElement): ReactElement => item)}
      <div className={`form-custom__submit`}>
        <button
          className={`form-custom__btn`}
          type="submit"
          disabled={!formValid}
        >
          {titel}
        </button>
      </div>
    </form>
  );
};

export { FormCustom };

const wrapper = compose(withFormState);

export default wrapper(FormCustom) as ComponentClass<PropsInsert>;
