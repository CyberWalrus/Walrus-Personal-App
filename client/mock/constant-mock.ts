import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
