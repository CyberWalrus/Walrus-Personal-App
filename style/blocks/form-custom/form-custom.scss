.form-custom {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: column;
  > * {
    margin-top: 2rem;
  }
}
