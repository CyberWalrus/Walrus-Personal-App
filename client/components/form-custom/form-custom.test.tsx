import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { FORM_TYPE, optionForm } from "../../mock/constant-mock";
import { FormCustom } from "./form-custom";

Enzyme.configure({ adapter: new Adapter() });

it(`FormCustom correctly render`, (): void => {
  const tree: Json = toJson(
    shallow(
      <FormCustom options={optionForm} formType={FORM_TYPE} titel={`test`} />,
    ),
  );
  expect(tree).toMatchSnapshot();
});
