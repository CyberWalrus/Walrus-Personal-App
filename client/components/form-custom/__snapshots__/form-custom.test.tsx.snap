// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCustom correctly render 1`] = `
<form
  className="form-custom"
  onSubmit={[Function]}
>
  <div
    className="form-custom__titel"
  >
    <h2>
      test
    </h2>
  </div>
  <div
    className="form-custom__message"
  />
  <InputCustom
    key="1"
    onChangeInput={[Function]}
    option={
      Object {
        "formTypes": Array [
          "SIGN_UP",
          "SIGN_IN",
        ],
        "hidenValue": "Email address",
        "name": "email",
        "placeHolder": "Email address",
        "type": "email",
      }
    }
  />
  <InputCustom
    key="2"
    onChangeInput={[Function]}
    option={
      Object {
        "formTypes": Array [
          "SIGN_UP",
          "SIGN_IN",
        ],
        "hidenValue": "Password",
        "name": "password",
        "placeHolder": "Password",
        "type": "password",
      }
    }
  />
  <div
    className="form-custom__submit"
  >
    <button
      className="form-custom__btn"
      disabled={false}
      type="submit"
    >
      test
    </button>
  </div>
</form>
`;
